src/test/java/com/mongodb/jdbc/MongoSQLResultSetMetaDataTest.java

```diff
@@ -20,7 +20,7 @@
 
     static {
         try {
-            resultSetMetaData = new MongoSQLResultSetMetaData(generateMongoJsonSchema());
```

Comment:
It may be useful to add a unit test here to confirm the expected sort order with isDatabaseMetadataResultSet set and unset.  It's similar to the integration tests, but more direct as opposed to going through the DatabaseMetaData or Statement calls.

Reply:
I'm sorry but I don't quite see what you are proposing here. Can you provide more details?
MongoDatabaseMetaDataTest tests Database metadata resultsets and will verify that the order is not alphabetical but follow the order of the spec. This class will verify that result sets columns for SQL statements are sorted alphabetically. Since, the decision to set or unset is not "our" but is based on the result set type, it seems enough and less confusing to only test the expected behaviour.
Or are you asking for a new test to check that the result set metadata columns are sorted alphabetically? I just checked and it does not seem to be here. This would be a good test to add in this file.

Reply:
Yes, the latter.  A test that confirms the sorting of the columns that we get from generateMongoJsonSchema() are what we are expecting.

Reply:
Yes, I think it makes sense. Even though, technically, this is tested because we have tests checking the database metadata order and tests checking SQL statement result sets order.
Added a unit test for this.